recipe/patches/0003-patch-zoneinfo_dir_override-to-point-to-our-tzdata.patch

```diff
@@ -0,0 +1,113 @@
+From 364f9fd8505a162a6cacd0223c3b55927804981c Mon Sep 17 00:00:00 2001
+From: "H. Vetinari" <h.vetinari@gmx.com>
+Date: Fri, 21 Jun 2024 12:41:07 +1100
+Subject: [PATCH 3/5] patch zoneinfo_dir_override to point to our tzdata
+
+---
+ libstdc++-v3/src/c++20/tzdb.cc | 78 ++++++++++++++++++++++++++++------
+ 1 file changed, 65 insertions(+), 13 deletions(-)
+
+diff --git a/libstdc++-v3/src/c++20/tzdb.cc b/libstdc++-v3/src/c++20/tzdb.cc
+index c7c7cc9de..1596a77b2 100644
+--- a/libstdc++-v3/src/c++20/tzdb.cc
++++ b/libstdc++-v3/src/c++20/tzdb.cc
+@@ -37,8 +37,12 @@
+ #include <mutex>      // mutex
+ #include <filesystem> // filesystem::read_symlink
+ 
+-#ifndef _AIX
+-# include <cstdlib>   // getenv
++#if defined(__linux__)
++# include <dlfcn.h>   // dladdr
++# include <cstdlib>   // free, malloc, realpath
++# include <cstring>   // memcpy, strlen
++#else defined(_WIN32)
++# include <windows.h>
+ #endif
+ 
+ #if defined __GTHREADS && ATOMIC_POINTER_LOCK_FREE == 2
+@@ -64,25 +68,73 @@
+ 
+ namespace __gnu_cxx
+ {
+-#ifdef _AIX
+-  // Cannot override weak symbols on AIX.
+-  const char* (*zoneinfo_dir_override)() = nullptr;
+-#else
```

Comment:
Let's keep this block or at least #error so that we don't trip if we build this.